src/Api/ApiResourceRegistrar.php

```diff
@@ -54,9 +59,18 @@ public function __construct(Router $api)
      * @param string $resourceName URI of resource
      * @param string $controller FQDN Class name of Controller, which contains action method
      * @param array $options options, passed to router on route registration
+     * @param string $modelClass Model to resolve binding
+     * @param string $modelName Model parameter name
+     *
+     * @throws \ReflectionException
      */
-    public function resource(string $resourceName, string $controller, array $options = [])
-    {
+    public function resource(
+        string $resourceName,
+        string $controller,
+        array $options = [],
+        string $modelClass = null,
+        string $modelName = null
```

Comment:
isn't **modelName** in common cases is same as **resourceName**? What do you think, can we use resource name when model name not passed? We can use Str::singular($resourceName) to retrieve singular form

Reply:
I see that we can leave this parameter empty, but, for example, when we want to get some 'participant' of class User, should we use **'participants/{user}'** or **'participants/{participant}'**? I know that we can configure it via parameters, but which behavior should be by default?

Reply:
When model name not passed we should use short name of given Model class. I think this behavior more clearly. How you said in common cases modelName will be as class short name

Reply:
When we pass model class we tells, that we want to get object of this class. ModelName means that we will be found this object with this give name. Answer on your question: you can use both of these routes, but you must remember that participant should be instance of user, and when you pass participant parameter, but your controller get User $user, you should do next router call 
$api->resource('participants', 'controller', [], Participant::class, 'user')